src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -568,6 +641,80 @@ export default class AircraftController {
     }
 
     /**
+     * Generate a unique `transponderCode`
+     *
+     * This method should only be run while building props for a
+     * soon-to-be-instantiated `AircraftModel`
+     *
+     * @for AircraftController
+     * @method _generateUniqueTransponderCode
+     * @return {number}
+     * @private
+     */
+    _generateUniqueTransponderCode() {
+        const minCodeValue = 1203;
+        const maxCodeValue = 7499;
+        const transponderCode = _random(minCodeValue, maxCodeValue);
+
+        if (!this._isValidTransponderCode(transponderCode)) {
+            // the value generated is already in use, recurse back through this method and try again
+            this._generateUniqueTransponderCode();
```

Comment:
Is there not a better way to do this and avoid recursion? Like maybe something like a randomly sorted array of all available ones, then pulling from that list? Seems like a simpler guarantee of uniqueness than generating random ones and then checking to see if they're unique. Particularly with the CIDs (since there's only 999 of them), once you get over a hundred or so aircraft, the frequency of a need to re-generate a new one because the random selection is actually in use is quite a bit greater. Just curious if there's another approach that might be better.